internal/engine/upper.go

```diff
@@ -1005,7 +1005,11 @@ func handleTiltfileLogAction(ctx context.Context, state *store.EngineState, acti
 	state.TiltfileCombinedLog = model.AppendLog(state.TiltfileCombinedLog, action, state.LogTimestamps, nil)
 }
 
-func handleAnalyticsOptAction(state *store.EngineState, action store.AnalyticsOptAction) {
+func handleAnalyticsOptAction(ctx context.Context, state *store.EngineState, action store.AnalyticsOptAction) {
+	// only logging on opt-in, because, well, opting out means the user just told us not to report data on them!
+	if action.Opt == analytics.OptIn {
+		tiltanalytics.Get(ctx).IncrIfUnopted("analytics.opt.in")
```

Comment:
it seems very sus to be sending requests from inside a reducer loop because this is holding the lock...would it be better to do this where the action gets fired?

Reply:
yeah, it's probably cleaner to do it where the action is fired.

I'm less certain about it being suspect to do this inside the reducer loop - we take pains to ensure that the analytics methods do not send requests synchronously, and I think that's a good contract to have for metrics reporting, so that we can easily add metrics without worrying about where we're doing it.

anyway, I've moved it to the http request handler since that more directly corresponds to the user action we want to record